vms/proposervm/block/block.go

```diff
@@ -40,6 +45,20 @@ type SignedBlock interface {
 	// Proposer returns the ID of the node that proposed this block. If no node
 	// signed this block, [ids.EmptyNodeID] will be returned.
 	Proposer() ids.NodeID
+
+	VRFSig() []byte
+
+	// VerifySignature validates the correctness of the VRF signature.
+	VerifySignature(*bls.PublicKey, []byte, ids.ID, uint32) bool
```

Comment:
q: should we consider passing `ctx *snow.Context` rather then chainID and networkID separately? Just to slim down the signature.
I don't have a strong opinion about this, just raising the point

Reply:
The `snow` package is currently not a dependency of the `block` package.
Do you forsee any future parameters that would be derived off `snow.Context` ? 
I can see that the key, networkID and chainID are pretty much "constants", whereas the `signature` is varying. maybe we should bundle these three up and store them in the VM ? 
```golang
type signatureVerificationContext struct {
    blsKey *bls.PublicKey
    chainID ids.ID
    networkID uint32
}
```